cobra/core/Model.py

```diff
@@ -720,3 +654,39 @@ def __exit__(self, type, value, traceback):
         """Pop the top context manager and trigger the undo functions"""
         context = self._contexts.pop()
         context.reset()
+
+
+def get_reactions(model, reactions):
```

Comment:
So, I agree this is good functionality, but would it be better suited for `dictlist`?
Something like `get_by_id`, but would accept an iterable, would check for an `id` attribute and search on that if appropriate, etc.? Then we wouldn't need a `get_metabolites` down the road.

Just a consideration, I'm fine with this as is